### ISSUE TITLE
Chakra should treat xbox as a low memory device

### ISSUE THREAD

**Issue by MikeHolman** [Microsoft/ChakraCore#4025]
In particular I think we might want to check GetProcessInformation for AvailableCommit.

**Comment by curtisman** [comment]
XBox shouldn't be special, and changing that won't make any different for xbox, as we only enable the tiny memory recycler heuristics looking at physical memory  <512.

We can probably remove looking at the device class totally.

**Comment by MikeHolman** [comment]
Yes I know we also consider the physical memory, that's why I didn't just make the change. It requires a bit more finesse than the title suggests. But also, it's why I suggested checking AvailableCommit (or some other mechanism to get the real commit limit). I didn't explicitly say it, but I thought we could do that instead of looking at device class.

Anyway, I think this is a severe issue, and if the full change is feature level work, I think we should do something targeted as a bug fix. Basically any website that is creating temporary objects somewhat often (or has some spike where they do) is going to face serious problems.

Check this out for an example:
```` js
let foo = [
	{x:5,y:10,z:20},
	{x:5,y:10,z:20},
	{x:6,y:10,z:25},
	{x:6,y:11,z:20},
	{x:7,y:10,z:20}
];

let start = new Date();
let a = false;
for (let i = 0; i < 10000000000; ++i)
{	
	if(I%5000 == 0)
	{
		a = JSON.stringify(foo) === JSON.stringify(foo);	
	}
}
print(new Date()-start);
print(a);
````

I see about 170MB commit while this is running. If I hack chakracore to give me the low memory heuristic I see memory drop to 27MB commit.

Yes this is a silly test and if you put a longer delay between these the memory will come down, but I don't think it's so outrageous to expect someone will create this much in temporary objects.
